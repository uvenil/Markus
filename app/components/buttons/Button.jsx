'use strict';

import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import FontIcon from 'material-ui/FontIcon';
import Constants from '../../utils/Constants';
import _ from 'lodash';

const Button = props => {
    const icon = props.icon ?
        <FontIcon
            className={'fa fa-fw fa-' + props.icon}
            style={{ marginLeft : props.labelPosition === 'before' || !props.label ? 0 : Constants.PADDING_X1, marginRight : props.labelPosition === 'after' || !props.label ? 0 : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE }} /> : undefined;

    const style = { minWidth : props.width, height : props.height, lineHeight : props.height + 'px', textAlign : props.align };

    if (props.borderless) {
        return (
            <FlatButton
                label={props.label}
                labelPosition={props.labelPosition}
                labelStyle={props.label ? { paddingLeft : Constants.PADDING_X1, paddingRight : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE, fontWeight : props.labelWeight === 'light' ? 300 : props.labelWeight === 'bold' ? 500 : 400, textTransform : 'none' } : {}}
                icon={icon}
                primary={props.color === 'primary'}
                secondary={props.color === 'secondary'}
                disabled={props.disabled}
                style={_.assign(style, props.style)}
                onTouchTap={props.onTouchTap} />
        );
    }

    return (
        <RaisedButton
            label={props.label}
            labelStyle={props.label ? { paddingLeft : Constants.PADDING_X1, paddingRight : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE, textTransform : 'none' } : {}}
            icon={icon}
            primary={props.color === 'primary'}
            secondary={props.color === 'secondary'}
            disabled={props.disabled}
            style={_.assign(style, props.style)}
            onTouchTap={props.onTouchTap} />
    );
};

Button.propTypes = {
    label         : React.PropTypes.string,
    labelPosition : React.PropTypes.oneOf([ 'before', 'after' ]),
    labelWeight   : React.PropTypes.oneOf([ 'light', 'normal', 'bold' ]),
    icon          : React.PropTypes.string,
    borderless    : React.PropTypes.bool,
    color         : React.PropTypes.oneOf([ 'default', 'primary', 'secondary' ]),
    width         : React.PropTypes.oneOfType([ React.PropTypes.number, React.PropTypes.string ]),
    height        : React.PropTypes.oneOfType([ React.PropTypes.number, React.PropTypes.string ]),
    align         : React.PropTypes.oneOf([ 'left', 'center', 'right' ]),
    disabled      : React.PropTypes.bool,
    style         : React.PropTypes.object,
    onTouchTap    : React.PropTypes.func
};

Button.defaultProps = {
    labelPosition : 'after',
    labelWeight   : 'bold',
    width         : Constants.BUTTON_MIN_WIDTH,
    height        : Constants.BUTTON_MIN_HEIGHT,
    borderless    : true,
    color         : 'default',
    align         : 'center',
    disabled      : false
};

export default Button;
