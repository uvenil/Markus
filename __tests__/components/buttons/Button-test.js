'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import Button from '../../../app/components/buttons/Button.jsx';

it('renders correctly', () => {
    const button = Renderer.create(
        <Button>Hello, World!</Button>
    ).toJSON();

    expect(button).toMatchSnapshot();
});
