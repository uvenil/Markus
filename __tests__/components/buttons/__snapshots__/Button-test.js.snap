exports[`test renders correctly 1`] = `
<button
  className="Button"
  onClick={[Function]}
  style={
    Object {
      "WebkitUserSelect": "none",
      "backgroundColor": "#f5f7fd",
      "borderColor": "#f5f7fd",
      "borderRadius": "4px",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "color": "#000",
      "cursor": "default",
      "fontFamily": undefined,
      "fontSize": undefined,
      "outline": "none",
      "padding": "4px 8px",
      "pointerEvents": "auto",
      "width": undefined,
    }
  }
  type="button">
  Hello, World!
</button>
`;
